DxDispatch/src/dxdispatch/DmlDispatchable.cpp

```diff
@@ -84,18 +121,39 @@ void FillBindingData(
             for (auto& source : sources)
             {
                 assert(source.resource != nullptr);
-                assert(source.resourceDesc != nullptr);
-
-                if (!std::holds_alternative<Model::BufferDesc>(source.resourceDesc->value))
+                if (isSerializedGraph)
                 {
-                    throw std::invalid_argument("DML operators only support buffer bindings");
+                    // Use SafeMultiply and CalculateSize only for DMLSerialized
+                    uint64_t offset = SafeMultiply(source.elementOffset, source.elementSizeInBytes);
+                    uint64_t sizeInBytes = CalculateSize(source.elementCount, source.elementSizeInBytes);
+
+                    bindingData.bufferBindings[bufferIndex].Offset = offset;
+                    bindingData.bufferBindings[bufferIndex].SizeInBytes = sizeInBytes - offset;
+
+                    // Validate buffer size
+                    if (offset + SafeMultiply(source.elementCount, source.elementSizeInBytes) > sizeInBytes)
+                    {
+                        throw std::invalid_argument(fmt::format(
+                            "Buffer size ({} bytes) is too small for the data ({} bytes) at offset {} bytes for binding point '{}'", 
+                            sizeInBytes,
+                            SafeMultiply(source.elementCount, source.elementSizeInBytes),
+                            offset,
+                            bindPointName));
+                    }
+                }
+                else
+                {                    
+                    if (!std::holds_alternative<Model::BufferDesc>(source.resourceDesc->value))
+                    {
+                        throw std::invalid_argument("DML operators only support buffer bindings");
+                    }
```

Comment:
Move this check also outside if/else. We would want to verify the resourceDesc for both dispatchable.

Reply:
resolved